modules/library/main/src/main/java/org/geotools/geometry/jts/coordinatesequence/CoordinateSequences.java

```diff
@@ -177,7 +178,7 @@ public static int coordinateDimension(CoordinateSequence seq) {
         // hack to handle issue that CoordinateArraySequence always reports
         // dimension = 3
         // check if a Z value is NaN - if so, assume dim is 2
-        if (seq instanceof CoordinateArraySequence) {
+        if (seq instanceof CoordinateArraySequence || seq instanceof LiteCoordinateSequence) {
```

Comment:
Not sure about this change, could you elaborate? Maybe related to the removal of the dimension guess in LiteCoordinateSequence?

Reply:
Yes, otherwise `LiteCoordinateSequeces` return a dimension of 3 and the GML encoder exports the `Z` values as `NaN`